keyboards/gmmk/pro/ansi/keymaps/coryginsberg/keymap.c

```diff
@@ -43,69 +27,148 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+      tap_code(KC_VOLU);
+    } else {
+      tap_code(KC_VOLD);
+    }
+    return true;
+}
+
+// Capslock, Scroll lock and Numlock indicator on Left side lights.
+void rgb_matrix_indicators_advanced_user(uint8_t led_min, uint8_t led_max) {
+    if (IS_HOST_LED_ON(USB_LED_SCROLL_LOCK)) {
+        rgb_matrix_set_color(LED_L1, RGB_GREEN);
+        rgb_matrix_set_color(LED_L2, RGB_GREEN);
+    }
+    if (IS_HOST_LED_ON(USB_LED_NUM_LOCK)) {
+        rgb_matrix_set_color(LED_L3, RGB_MAGENTA);
+        rgb_matrix_set_color(LED_L4, RGB_MAGENTA);
+    }
+    if (IS_HOST_LED_ON(USB_LED_CAPS_LOCK)) {
+        rgb_matrix_set_color(LED_L5, RGB_RED);
+        rgb_matrix_set_color(LED_L6, RGB_RED);
+        rgb_matrix_set_color(LED_L7, RGB_RED);
+        rgb_matrix_set_color(LED_CAPS, RGB_RED);
+    }
+    if (keymap_config.no_gui) {
+        rgb_matrix_set_color(LED_LWIN, RGB_RED);  //light up Win key when disabled
+    }
+    switch(get_highest_layer(layer_state)){  // special handling per layer
+        case 2:  // on Fn layer select what the encoder does when pressed
+            for (uint8_t i=0; i<ARRAYSIZE(LED_SIDE_LEFT); i++) {
+                rgb_matrix_set_color(LED_SIDE_LEFT[i], RGB_OFF);
+                rgb_matrix_set_color(LED_SIDE_RIGHT[i], RGB_OFF);
+            }
+            for (uint8_t i=0; i<ARRAYSIZE(LED_LIST_FUNCROW); i++) {
+                rgb_matrix_set_color(LED_LIST_FUNCROW[i], RGB_OFF);
+            }
+            rgb_matrix_set_color(LED_FN, RGB_GREEN); //FN key
+            rgb_matrix_set_color(LED_END, RGB_RED);
+            rgb_matrix_set_color(LED_1, RGB_RED);
+            rgb_matrix_set_color(LED_INS, RGB_YELLOW);
+            rgb_matrix_set_color(LED_BSLS, RGB_RED);
+            for (uint8_t i=0; i < ARRAYSIZE(LED_LIST_ARROWS); i++) {
+                rgb_matrix_set_color(LED_LIST_ARROWS[i], RGB_BLUE);
+            }
+            for (uint8_t i=0; i< ARRAYSIZE(LED_LIST_WASD); i++) {
+                rgb_matrix_set_color(LED_LIST_WASD[i], RGB_PURPLE);
+            }
+            break;
+        case 3:  // on Fn layer select what the encoder does when pressed
+            rgb_matrix_set_color_all(RGB_OFF);
+            for (uint8_t i=0; i < ARRAYSIZE(LED_LIST_NUMPAD); i++) {
+                rgb_matrix_set_color(LED_LIST_NUMPAD[i], RGB_GREEN);
+                rgb_matrix_set_color(LED_LIST_NUM_OTHERS[i], RGB_YELLOW);
+            }
+            for (uint8_t i=0; i < ARRAYSIZE(LED_LIST_NUMPAD_ACTIONS); i++) {
+                rgb_matrix_set_color(LED_LIST_NUMPAD_ACTIONS[i], RGB_BLUE);
+            }
+            for (uint8_t i=0; i < ARRAYSIZE(LED_SIDE_LEFT); i++) {
+                rgb_matrix_set_color(LED_SIDE_LEFT[i], RGB_WHITE);
+                rgb_matrix_set_color(LED_SIDE_RIGHT[i], RGB_WHITE);
+            }
+            break;
+    }
+}
+
+void suspend_power_down_user(void) {
+    rgb_matrix_set_suspend_state(true);
+}
+
+void suspend_wakeup_init_user(void) {
+    rgb_matrix_set_suspend_state(false);
+}
```

Comment:
If `RGB_DISABLE_WHEN_USB_SUSPENDED` is defined, these are not needed, as it's handled in core. 
```suggestion
```